design/20230719-cabundle-reference.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Certificate Request CRD
+authors:
+  - "@AppalaKarthik"
+  - "@sankalp-at-github"
+creation-date: 2023-07-19
+last-updated: 2023-07-19
+#status: implementable
+---
+
+# Design: Certficate Authority Bundle Reference Using Secrets, Configmaps
+
+
+<!-- toc -->
+- [Problem Statement](#problem-statement)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [Graduation Criteria](#graduation-criteria)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Problem Statement
+
+In Issuer/ClusterIssuer resources caBundle field must be specified as a base64 encoded string.
+
+```
+spec:
+  venafi:
+    tpp:
+      caBundle: <B64_ENCODED_STRING>
+      credentialsRef:
+        name: tpp-token
+      url: https://my-server.com/vedsdk/
+    zone: Certificates\public
+```
+
+This becomes a configuration management issue when many issuers are there in a cluster using the same TPP endpoint, the same CA information has to be copied in all the issuers and there can be mistakes made as this would be manual.
+
+## Summary
+
+Ability to reference caBundle from the contents of secrets, configmap.
+
+## Motivation
+The proposed change would encourage the way caBundle is referenced in kubernetes native way.
+
+### Goals
+
+To be able to refer caBundle using secret.
+To be able to refer caBundle using configmap.
```

Comment:
I think we should also specify that we want to stay up-to-date with the latest version of the resource (if the resource contents change, we want to start trusting the new CA).
TODO: check if https://github.com/cert-manager/cert-manager/pull/5387 does this properly & how it does this.

Reply:
Agreed. I think #5387 does this correctly for secrets, ref. https://github.com/cert-manager/cert-manager/blob/27fb916f64da0557ad7c267125c0b1da34cf0817/pkg/controller/issuers/checks.go#L91-L96 (and the same is implemented for cluster issuers).